chia/rpc/wallet_rpc_api.py

```diff
@@ -924,6 +925,17 @@ async def create_offer_for_ids(self, request):
         offer: Dict[str, int] = request["offer"]
         fee: uint64 = uint64(request.get("fee", 0))
         validate_only: bool = request.get("validate_only", False)
+        type_dict_str: Optional[Dict[str, str]] = request.get("type_dict", None)
+
+        # This type_dict construction is to maintain backward compatibility where everything is assumed to be a CAT
+        type_dict: Dict[str, str] = {}
```

Comment:
The `type_dict` type definition differs from what TradeManager's `create_offer_for_ids` expects: `Dict[bytes32, AssetType]`